app/src/main/java/com/xmartlabs/template/controller/ServiceAndDatabaseController.java

```diff
@@ -0,0 +1,140 @@
+package com.xmartlabs.template.controller;
+
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+
+import com.raizlabs.android.dbflow.sql.language.property.Property;
+import com.xmartlabs.template.helper.function.BiFunction;
+import com.xmartlabs.template.helper.function.Function;
+import com.xmartlabs.template.model.EntityWithId;
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Flowable;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@SuppressWarnings({"unused", "WeakerAccess"})
+public abstract class ServiceAndDatabaseController<Id, D extends EntityWithId<Id>, C> extends Controller {
```

Comment:
The decision was taken because maybe an entityController does not implement all functions. Also these functions should not be public, for example imagine that you have a `UserController` which has a `getUser(id)` method. This function will use the `getEntity(ServiceCall method)` but this method should not be public